charts/consul/templates/terminating-gateways-service.yaml

```diff
@@ -0,0 +1,31 @@
+{{- if and .Values.terminatingGateways.enabled }}
+
+{{- $root := . }}
+{{- $defaults := .Values.terminatingGateways.defaults }}
+
+{{- range .Values.terminatingGateways.gateways }}
+
+{{- $service := .service }}
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ template "consul.fullname" $root }}-{{ .name }}
+  namespace: {{ $root.Release.Namespace }}
+  labels:
+    app: {{ template "consul.name" $root }}
+    chart: {{ template "consul.chart" $root }}
+    heritage: {{ $root.Release.Service }}
+    release: {{ $root.Release.Name }}
+    component: terminating-gateways
+spec:
+  selector:
+    app: {{ template "consul.name" $root }}
+    release: "{{ $root.Release.Name }}"
+    component: terminating-gateways
+  type: ClusterIP
+  ports:
+    - port: 80
+      targetPort: 8443
```

Comment:
This port will not get used under normal operation. It is needed however in order to meet the spec for a service. Otherwise Helm/K8s will throw a fit.